### PR TITLE
MM-36990: Use upsert for system key update

### PR DESCRIPTION
**Pull request by agnivade**
We make the app migration idempotent by using DB native upsert.

https://mattermost.atlassian.net/browse/MM-36990

```release-note
NONE
```

### CODE REVIEW

**Comment by agnivade** [comment]
We could also do an insert and check for the return status and ignore a conflict. But this seemed to be more natural to me.

**Review by mickmister — APPROVED** [review]
LGTM

File: store/sqlstore/system_store.go [hunk 1]
```diff
@@ -45,31 +46,39 @@ func (s SqlSystemStore) Save(system *model.System) error {
...
 	} else {
-		if err := s.GetMaster().Insert(system); err != nil {
-			return errors.Wrapf(err, "failed to save system property with name=%s", system.Name)
-		}
+		query = query.SuffixExpr(sq.Expr("ON CONFLICT (name) DO UPDATE SET Value = ?", system.Value))
+	}
+
+	queryString, args, err := query.ToSql()
+	if err != nil {
+		return errors.Wrap(err, "status_tosql")
```
Comment (streamer45) [inline store/sqlstore/system_store.go:62]:
```suggestion
		return errors.Wrap(err, "system_tosql")
```

**Review by streamer45 — APPROVED** [review]
Nice :+1:

**Pull request merged**